samples/apply-stretch-renderer/src/main/java/com/esri/arcgismaps/sample/applystretchrenderer/components/ApplyStretchRendererViewModel.kt

```diff
@@ -0,0 +1,260 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.applystretchrenderer.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.RasterLayer
+import com.arcgismaps.mapping.symbology.raster.MinMaxStretchParameters
+import com.arcgismaps.mapping.symbology.raster.PercentClipStretchParameters
+import com.arcgismaps.mapping.symbology.raster.StandardDeviationStretchParameters
+import com.arcgismaps.mapping.symbology.raster.StretchParameters
+import com.arcgismaps.mapping.symbology.raster.StretchRenderer
+import com.arcgismaps.raster.Raster
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.applystretchrenderer.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+
+private const val DEFAULT_MIN = 10.0
+private const val DEFAULT_MAX = 150.0
+private const val DEFAULT_PERCENT_MIN = 0.0
+private const val DEFAULT_PERCENT_MAX = 50.0
+private const val DEFAULT_STD_DEVIATION_FACTOR = 0.5
+
+/**
+ * ViewModel for the "Apply stretch renderer" sample.
+ */
+class ApplyStretchRendererViewModel(private val app: Application) : AndroidViewModel(app) {
+    private var appliedMinValue = DEFAULT_MIN
+    private var appliedMaxValue = DEFAULT_MAX
+    private var appliedPercentMin = DEFAULT_PERCENT_MIN
+    private var appliedPercentMax = DEFAULT_PERCENT_MAX
+    private var appliedStdDeviationFactor = DEFAULT_STD_DEVIATION_FACTOR
+    private var appliedStretchType = StretchType.MinMax
+
+    // The map used in the sample, with imagery basemap
+    val arcGISMap: ArcGISMap = ArcGISMap(BasemapStyle.ArcGISImageryStandard)
+
+    // MapViewProxy used to set viewpoint after layer loads
+    val mapViewProxy = MapViewProxy()
+
+    // Message dialog view model for error handling
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Provision path for local offline resources (raster-file/Shasta.tif)
+    private val provisionPath: String by lazy {
+        app.getExternalFilesDir(null)?.path.toString() + File.separator +
+                app.getString(R.string.apply_stretch_renderer_app_name)
+    }
+
+    // The raster data (Shasta.tif) should be downloaded to external storage on launch
+    private val raster: Raster by lazy {
```

Comment:
You could add in the correct path here: 

```suggestion
    // The raster data (raster-file/Shasta.tif) should be downloaded to external storage on launch
    private val raster by lazy {
```